doc/topics/packaging/testing.rst

```diff
@@ -0,0 +1,150 @@
+.. _pkging-testing:
+
+================
+Testing packages
+================
+
+The package test suite
+======================
+
+The salt repo provides a test suite for testing basic functionality of our
+packages at ``<repo-root>/pkg/tests/``. You can run the install, upgrade, and
+downgrade tests. These tests run automatically on most PRs that are submitted
+against Salt.
+
+
+.. warning::
+
+    These tests make destructive changes to your system because they install the
+    built packages onto the system. They may also install older versions in the
+    case of upgrades or downgrades. To prevent destructive changes, run the
+    tests in an isolated system, preferably a virtual machine.
+
+Setup
+=====
+In order to run the package tests, the `relenv
+<https://github.com/saltstack/relative-environment-for-python>`_ onedir and
+built packages need to be placed in the correct locations.
+
+Place all salt packages for the applicable testing version in
+``<repo-root>/pkg/artifacts/``, and the onedir must be located under
+``<repo-root>/artifacts/``. Additionally, to ensure complete parity with Salt's
+CI/CD suite, place the ``nox`` virtual environment in
+``<repo-root>/.nox/test-pkgs-onedir`` The following are a few ways this can be
+accomplished easily.
+
+You can ensure parity by installing the package test suite through a few
+possible methods:
+
+* Using ``tools``
+* Downloading individually
+
+Using ``tools``
+---------------
+Salt has preliminary support for setting up the package test suite in the
+``tools`` command suite that is located under ``<repo-root>/tools/testsuite/``.
+This method requires the Github CLI tool ``gh`` to be properly configured for
+interaction with the salt repo.
+
+#. Install the dependencies using this command:
+
+    .. code-block:: bash
+
+       pip install -r requirements/static/ci/py{python_version}/tools.txt
+
+#. Download and extract the artifacts with this ``tools`` command:
+
+
+    .. code-block:: bash
+
+        tools ts setup --platform {linux|darwin|windows} --slug
+        <operating-system-slug> --pr <pr-number> --pkg
+
+    The most common use case is to test the packages built on a CI/CD run for a
+    given PR. To see the possible options for each argument, and other ways to
+    utilize this command, use the following.
```

Comment:
**Suggestion (punctuation, non-blocking):** Maybe change this period to a colon.